File: DataStructure-and-Algorithm/Search/Binary_Search/binary_search.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid + 1
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1

l = []
print("Enter 10 numbers :")
for i in range(10):
    n1 = int(input())
    l.append(n1)

l.sort()
print("Sorted list:", l)

n = int(input("Enter the number to be searched : "))
p = binary_search(l, n)

if p != -1:
    print(f"Found at {p} position")
else:
    print("Not Found")

File: DataStructure-and-Algorithm/Search/Binary_Search/binary_search.c
#include <stdio.h>

int search(int [],int,int);
int main(void)
{

    int n,i,a[100],e,res;
    printf("Enter the number of Elements : ");
    scanf("%d",&n);
    printf("Enter Elements of Array in Ascending order : ");
    for(i=0;i<n;++i)
    {
        scanf("%d",&a[i]);
    }
    printf("Enter element to search : ");
    scanf("%d",&e);
    
    res=search(a,n,e);
    
    if(res!=-1)
        printf("Element found at position %d",res+1);
    else
        printf("Element is not found!"); 
    return 0;
}
 
int search(int a[],int n,int e)
{
    int f,l,m;
    f=0;
    l=n-1;
    
    while(f<=l)
    {
        m=(f+l)/2;
        if(e==a[m])
            return(m);
        else
            if(e>a[m])
                f=m+1;
            else
                l=m-1;
    }
    
    return -1;
}